x-pack/legacy/plugins/ml/public/application/routing/routes/timeseriesexplorer.tsx

```diff
@@ -213,6 +218,11 @@ export const TimeSeriesExplorerUrlStateManager: FC<TimeSeriesExplorerUrlStateMan
 
   const boundsMinMs = bounds?.min?.valueOf();
   const boundsMaxMs = bounds?.max?.valueOf();
+
+  const [selectedForecastIdProp, setSelectedForecastIdProp] = useState<string | undefined>(
+    appState?.mlTimeSeriesExplorer?.forecastId
```

Comment:
question: why the init value is not `selectedForecastId`?

Reply:
In this case I wanted to make sure it gets the initial value from the URL/appState. `selectedForecastId` could be `undefined` if `isJobChange` is `true`.